mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala

```diff
@@ -292,15 +292,16 @@ class KMeans private (
       }.reduceByKey { case ((sum1, count1), (sum2, count2)) =>
         axpy(1.0, sum2, sum1)
         (sum1, count1 + count2)
-      }.collectAsMap()
+      }
 
       bcCenters.destroy(blocking = false)
 
       // Update the cluster centers and costs
       converged = true
-      totalContribs.foreach { case (j, (sum, count)) =>
+      totalContribs.map { case (j, (sum, count)) =>
```

Comment:
This is OK, but I think I'd keep the distributed operations together above instead.